src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -575,19 +576,24 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                 // in the factory method call for this symbol.
                 return ((ForEachLoopBinder)this.binder).InferCollectionElementType(diagnostics, _collection);
             }
+
+            internal override SyntaxNode ForbiddenZone => _collection;
         }
 
         /// <summary>
         /// Symbol for an out variable local that might require type inference during overload resolution.
         /// </summary>
         private class PossiblyImplicitlyTypedOutVarLocalSymbol : SourceLocalSymbol
         {
+            private readonly CSharpSyntaxNode _containingInvocation;
+
             public PossiblyImplicitlyTypedOutVarLocalSymbol(
                 Symbol containingSymbol,
                 Binder binder,
                 TypeSyntax typeSyntax,
                 SyntaxToken identifierToken,
-                LocalDeclarationKind declarationKind)
+                LocalDeclarationKind declarationKind,
+                CSharpSyntaxNode containingInvocation)
```

Comment:
Actually, I'll add that to this PR momentarily.

---
In reply to: [73925172](https://github.com/dotnet/roslyn/pull/12991#discussion_r73925172) [](ancestors = 73925172,73918408)